X바 - Y바의 99% 신뢰구간을 구하여라

In [1]:
a = (3.673 - 3.547) + 2.575*(0.7235+0.6089)
b = (3.673 - 3.547) - 2.575*(0.7235+0.6089)

In [7]:
aa = round(a,3)
bb = round(b,3)

In [6]:
aa

3.557

In [8]:
print("[" , aa , "," , bb , "]")

[ 3.557 , -3.305 ]


In [21]:
math.sqrt(9)

3.0

In [24]:
Z = (3.673 - 3.547)/(math.sqrt((0.7235**2 / 226)+(0.6089**2 / 247)))

In [25]:
Z

2.039377052705446

In [1]:
m = 216.07 - 219.82

In [2]:
m

-3.75

In [19]:
z = math.sqrt(46.49**2/58 + 60.65**2/49)

In [20]:
z

10.598772881515478

In [10]:
a = m - z
aa = round(a,3)
b = m + z
bb = round(b,3)
print('[',aa,',',bb,']')

[ -24.524 , 17.024 ]


In [11]:
math.sqrt(46.49**2/58 + 60.65**2/49)

10.598772881515478

In [12]:
-3.75/10.6

-0.3537735849056604

P-value < alpha => h0 reject
P-vlaue > alpha => 

In [23]:
p = m/z
pp = round(p,3)
print(pp)

-0.354


p-value를 구하기 위해선 - P-value는 넓이이다. 확률

검정 통계량을 알아야한다.

In [34]:
m1 = 120
m2 = 70
n1 = 12
n2 = 11
s1 = 10
s2 = 8

In [35]:
Sp_sq2 = (s1**2 * (n1 - 1) + s2**2 * (n2 - 1))/(n1 + n2 - 2)
Sp = math.sqrt(Sp_sq2)
print(round(Sp_sq2,3),'(',Sp_sq2,')')
print(round(Sp,3),'(',Sp,')')

82.857 ( 82.85714285714286 )
9.103 ( 9.102589898327995 )


In [45]:
T =(120-70) / (Sp*math.sqrt(1/n2 + 1/n1))
print(T)

13.159157202908345


P-Value 찾는거 해보기 (넓이는 통해서 표 없이)

---

## 2주차_2차시_09/12

In [8]:
a = math.exp(1)

In [9]:
a

2.718281828459045

In [4]:
# 정규분포 확률밀도함수
def normal_distribution(x,s,m):
    k = -0.5*(((x-m)/(s))**2)
    fx = (1/(s*math.sqrt(2*math.pi)))*math.exp(k)
    return fx

In [5]:
import numpy as np

In [7]:
# x = np.linspace(-20,20)
# y = normal_distribution(x,3,0)

In [35]:
# 예제 1.9
m1 = 49.39
m2 = 37.04
s1 = 317.456
s2 = 172.904
n1 = 12
n2 = 12
# 가설
# h0 : x1 - x2 > 0
# h1 : x1 - x2 <= 0

c = 2.528
df = 22
sp2 = (s1*s1*11 + s2*s2*11)/(df)
sp = math.sqrt(sp2)
print('sp : ',round(sp,3))
T = (m1 - m2)/(math.sqrt(1/12 + 1/12)*sp)
print('T :',round(T,3))
# -> h0 not reject

sp :  255.611
T : 0.118


In [36]:
#T검정
T = (m1 - m2) / (sp * math.sqrt(1/n1 + 1/n2))
sp = math.sqrt(sp2)
sp2 = (s1**2*(n1-1) + s2**2*(n2-1))/(df)
df = n1+n2-2

In [40]:
print(' T : ',round(T,3),'\t(',T,')')
print('sp : ',round(sp,3),'\t(',sp,')')

 T :  0.118 	( 0.11834851509035108 )
sp :  255.611 	( 255.61113546948616 )


In [42]:
class t_verification:
    T = (m1 - m2) / (sp * math.sqrt(1/n1 + 1/n2))
    sp = math.sqrt(sp2)
    sp2 = (s1**2*(n1-1) + s2**2*(n2-1))/(df)
    df = n1+n2-2
    def T(n1,n2,m1,m2,s1,s2) : return T
    def sp(n1,n2,m1,m2,s1,s2): return sp

In [44]:
t_verification.T(24,24,2.567,1.767,1.3027,1.0349)

0.11834851509035108

In [45]:
t_verification.T(12,12,49.39,37.04,317.453,172.904)

0.11834851509035108

### 예제 1-10 b)번 문제
1)    h0 : m1 - m2 = 0 | h1 : m1 - m2 != 0
2)    a = 0.01 < p-value = 0.023
3)    h0 not reject
4) 장흥지와 숭뢰지의 BOD함유량이 같다고 할 수 있다.

### 예제 1-10 c)번 문제
1) h0 : m1 - m2 <= 0 | h1 : m1 - m2 > 0
2) a = 0.01 < p-value = 0.0115
3) h0 not reject
4) 장흥지의 BOD함유량이 숭뢰지 보다 높다고 할 수 없다.

### 예제 1-7
estat.me 에서 계산 했음
1) h0 : m1 - m2 >= 0 | h1 : m1 - m2 < 0
2) a = 0.05 < p-value = 0.0635
3) h0 not reject
4) 6일의 금요일과 13일의 금요일은 차이가 없다고 할 수 있다.

---

## 3주차_1차시_09/15

In [9]:
import math
import numpy as np

### 예제 1-13

In [13]:
x = [12,29,16,37,28,15] 
y = [10,28,17,35,25,16] ## 이렇게 두개의 표본을
d = [] ## 뺌으로 인해 하나의 집단으로 계산가능해진다.
for i in range(len(x)):
    d.append(x[i] - y[i])
print(d)

[2, 1, -1, 2, 3, -1]


In [14]:
dm = np.mean(d)
print(dm)

1.0


In [15]:
d_mi = []
for i in range(len(d)):
    dmi = d[i] - dm
    d_mi.append(dmi**2)
    d_mi2 = np.sum(d_mi)
sd2 = d_mi2/5
sd = math.sqrt(sd2)
T = (dm - 0) / (sd/math.sqrt(6))
print('T : ',round(T,3))

T :  1.464


### 예제 1-14

In [19]:
x = [70,80,72,76,76,76,72,78,82,64,74,92,74,68,84]
y = [68,72,62,70,58,66,68,52,64,72,74,60,74,72,74]

In [24]:
d = []
for i in range(len(x)):
    d.append(x[i]-y[i])
n = len(d)
print(d)
print(n)

[2, 8, 10, 6, 18, 10, 4, 26, 18, -8, 0, 32, 0, -4, 10]
15


In [23]:
d_mean = np.mean(d)
print(d_mean)

8.8


In [26]:
dmi = []
for i in range(len(d)):
    d_minus = d[i] - d_mean
    dmi.append(d_minus**2)

sd2 = np.sum(dmi) / (n-1)

In [27]:
sd = math.sqrt(sd2)
T = (d_mean - 0) / (sd/math.sqrt(n))
print('T : ',round(T,3))

T :  3.105


## 3주차 2차시

### 예제 1-23

대응이 더 합리적으로 보인다.

In [4]:
year_2001 = [155,176,191,178,173,181,174,178,159,172,177,173,177,187,181]
year_2007 = [181,198,227,198,186,194,195,219,182,186,210,210,195,221,216]

In [6]:
year_temp = []
for i in range(len(year_2001)):
    year_temp.append(year_2001[i]-year_2007[i])
print(year_temp)

[-26, -22, -36, -20, -13, -13, -21, -41, -23, -14, -33, -37, -18, -34, -35]


In [11]:
y_mean = np.mean(year_temp)
y_mean

-25.733333333333334

In [12]:
y_vus = []
for i in range(len(year_temp)):
    wnd = year_temp[i] - y_mean
    y_vus.append(wnd**2)
sd2 = np.sum(y_vus)/14

In [13]:
sd = math.sqrt(sd2)

In [15]:
T = (y_mean - 0) / (sd/math.sqrt(15))
print('T : ',round(T,3))

T :  -10.46


$$\alpha = 0.01 \space \space 이면 \space \space R = -2.624 $$

$$T = -10.46 \space \space 이고 \space \space R = -2.624$$ $$이므로$$ $$H_0 를 \space 기각한다. \rightarrow 물가지수가\space 높다고\space 할\space 수\space 있다.$$

---

회귀분석의 목적 = 모델링 -> 가장 적합한 식을 찾는 것.

피타고리안 승률
$$ P = \frac{W^n}{W^n+L^n}$$
$$n = 2  \space  를\space 많이\space 사용한다.$$

$$n=1.83을 많이 넣기도 한다.$$

In [21]:
W = 10 #win
L = 10 #lose
n = 10 #1~3사이 임의의 수

In [22]:
P = (W**n) / (W**n + L**n)

In [23]:
import pandas as pd

In [30]:
pd.read_excel("C:\\Users\\SeungHyeon\\Desktop\\학교\\실험설계및분석\\[미션02][야구][팀][1982-2022].xlsx")

ID    시즌  순위   팀명    AVG    G    PA    AB    R     H  ...  IBB  HBP  \
0      1  2001   1   롯데  0.280  133  5308  4521  718  1265  ...   39   60   
1      2  2001   2  KIA  0.278  133  5189  4542  683  1263  ...   21   56   
2      3  2001   3   삼성  0.277  133  5280  4553  739  1262  ...   27   95   
3      4  2001   4   두산  0.276  133  5303  4583  732  1266  ...   27   60   
4      5  2001   5   LG  0.276  133  5319  4521  669  1247  ...   24   52   
..   ...   ...  ..  ...    ...  ...   ...   ...  ...   ...  ...  ...  ...   
189  190  2022   6   두산  0.255  144  5510  4891  638  1249  ...   16   71   
190  191  2022   7  SSG  0.254  144  5642  4893  720  1245  ...   22   95   
191  192  2022   8   KT  0.254  144  5547  4864  631  1237  ...   21   49   
192  193  2022   9   키움  0.252  144  5655  4953  621  1250  ...   26   92   
193  194  2022  10   한화  0.245  144  5513  4878  564  1196  ...   13   52   

       SO  GDP    SLG    OBP    OPS   MH   RISP  PH-BA  
0     817   89  0.425  0.365  0.790  133  0.288  0.244  
1     774  129  0.428  0.353  0.781  133  0.272  0.225  
2     814  111  0.447  0.360  0.807  133  0.267  0.223  
3     697  116  0.421  0.359  0.780  133  0.278  0.163  
4     759  100  0.391  0.363  0.754  133  0.280  0.263  
..    ...  ...    ...    ...    ...  ...    ...    ...  
189  1007  116  0.365  0.324  0.689  144  0.266  0.246  
190  1075   96  0.396  0.336  0.732  143  0.272  0.228  
191  1183  105  0.375  0.330  0.705  144  0.239  0.232  
192  1032   89  0.364  0.333  0.697  143  0.243  0.182  
193  1264   92  0.352  0.319  0.671  144  0.237  0.211  

[194 rows x 29 columns]

In [29]:
!pip install xlrd
!pip install openpyxl

     ---------------------------------------- 0.0/96.5 kB ? eta -:--:--
     -------------------------------------- - 92.2/96.5 kB 2.6 MB/s eta 0:00:01
     ---------------------------------------- 96.5/96.5 kB 1.8 MB/s eta 0:00:00
     ---------------------------------------- 0.0/250.0 kB ? eta -:--:--
     -------------- ------------------------ 92.2/250.0 kB 2.6 MB/s eta 0:00:01
     ---------------------------------- --- 225.3/250.0 kB 2.8 MB/s eta 0:00:01
     -------------------------------------- 250.0/250.0 kB 2.2 MB/s eta 0:00:00


In [33]:
pd.DataFrame({'win' : [74,69,65,64,60,62,58,51,53,51],
             'draw': [2,3,2,1,2,2,0,6,1,3],
             'lose': [47,55,53,57,57,59,64,66,71,78],
             'oow' : [0.612,0.556,0.551,0.529,0.513,0.512,0.475,0.436,0.427,0.395]})

win  draw  lose    oow
0   74     2    47  0.612
1   69     3    55  0.556
2   65     2    53  0.551
3   64     1    57  0.529
4   60     2    57  0.513
5   62     2    59  0.512
6   58     0    64  0.475
7   51     6    66  0.436
8   53     1    71  0.427
9   51     3    78  0.395

---

# 2장 일원배치 분산분석

## 예제 2-1

고정 모형이냐 랜덤 모형이냐에 따라 분석이 달라짐. 
이 예제는 고정 모형으로 보임.
이 판단을 하는게 상당히 중요함

랜덤하게 해야함. 순차적으로 하면 익숙해지면서 정확한 분석이 힘들어진다.

In [7]:
import numpy as np
import pandas as pd

In [5]:
ex2_1 = pd.DataFrame({'125' : [12,15,16,17],
                     '150' : [18,22,26,30],
                     '175' : [15,26,43,20],
                     '200' : [39,46,49,38]})

In [6]:
ex2_1

125  150  175  200
0   12   18   15   39
1   15   22   26   46
2   16   26   43   49
3   17   30   20   38

In [42]:
#평균
Y1 = np.mean(ex2_1['125'])
Y2 = np.mean(ex2_1['150'])
Y3 = np.mean(ex2_1['175'])
Y4 = np.mean(ex2_1['200'])
Y = np.mean([Y1,Y2,Y3,Y4])

print('Y1 : ',Y1,'\nY2 : ',Y2,'\nY3 : ',Y3,'\nY4 : ',Y4,'\nY : ',Y)

Y1 :  15.0 
Y2 :  24.0 
Y3 :  26.0 
Y4 :  43.0 
Y :  27.0


In [43]:
#잔차
Y_1 = list(list(ex2_1['125'])-Y1)
Y_2 = list(list(ex2_1['150'])-Y2)
Y_3 = list(list(ex2_1['175'])-Y3)
Y_4 = list(list(ex2_1['200'])-Y4)

print('Y_1 : ',Y_1,'\nY_2 : ',Y_2,'\nY_3 : ',Y_3,'\nY_4 : ',Y_4)

Y_1 :  [-3.0, 0.0, 1.0, 2.0] 
Y_2 :  [-6.0, -2.0, 2.0, 6.0] 
Y_3 :  [-11.0, 0.0, 17.0, -6.0] 
Y_4 :  [-4.0, 3.0, 6.0, -5.0]


In [47]:
def sum22(list):
    K = list
    sum = 0
    for i in range(4):
        sum = sum + K[i]**2
    return sum  

In [62]:
Y22 = list([sum22(Y_1),sum22(Y_2),sum22(Y_3),sum22(Y_4)])
SSE = np.sum(Y22)
print(SSE)

626.0


In [68]:
#처리효과
Yt_1 = Y1-Y
Yt_2 = Y2-Y
Yt_3 = Y3-Y
Yt_4 = Y4-Y

print('Yt_1 : ',Yt_1,'\nYt_2 : ',Yt_2,'\nYt_3 : ',Yt_3,'\nYt_4 : ',Yt_4)

Yt_1 :  -12.0 
Yt_2 :  -3.0 
Yt_3 :  -1.0 
Yt_4 :  16.0


In [76]:
Yt22 = [Yt_1**2,Yt_2**2,Yt_3**2,Yt_4**2]
SStr = np.sum(Yt22)*4
print(SStr)

1640.0


In [73]:
#총편차
Ya_1 = list(list(ex2_1['125'])-Y)
Ya_2 = list(list(ex2_1['150'])-Y)
Ya_3 = list(list(ex2_1['175'])-Y)
Ya_4 = list(list(ex2_1['200'])-Y)

print('Ya_1 : ',Ya_1,'\nYa_2 : ',Ya_2,'\nYa_3 : ',Ya_3,'\nYa_4 : ',Ya_4)

Ya_1 :  [-15.0, -12.0, -11.0, -10.0] 
Ya_2 :  [-9.0, -5.0, -1.0, 3.0] 
Ya_3 :  [-12.0, -1.0, 16.0, -7.0] 
Ya_4 :  [12.0, 19.0, 22.0, 11.0]


In [74]:
Ya22 = [sum22(Ya_1),sum22(Ya_2),sum22(Ya_3),sum22(Ya_4)]
SST = np.sum(Ya22)
print(SST)

2266.0


In [88]:
print(f'SST({SST}) = SStr({SStr}) + SSE({SSE}) = {SStr+SSE}')

SST(2266.0) = SStr(1640.0) + SSE(626.0) = 2266.0


In [86]:
MSE = round(626/12,2)
MSR = round(1640/3,2)
MST = round(2266/15,2)
print('MST = ',MST,'\nMSE = ',MSE,'\nMSR = ',MSR)

MST =  151.07 
MSE =  52.17 
MSR =  546.67


In [87]:
F = MSR/MSE
F

10.478627563733946

In [90]:
F3_12 = 3.49
F3_12

3.49

In [91]:
11664*16

186624

In [92]:
np.sqrt(186624)

432.0

In [101]:
sum = 0
for i in range(4):
    for j in range(4):
        sum = sum + ex2_1.loc[i][j]**2
print(sum)

13930


In [102]:
13930-11664

2266

---

## 10/06

p.40 - (c) = 6단계

p.41 - (f)
0. 제일 먼저 동질성부터 확인
1. 가성 설립
$$H_0 : \mu_1 - \mu_2 = 0$$

2. P-value = 0.141

3. 유의수준 5%에서 검정 

4. 검정 결과
$$H_0 \space \space not \space \space reject$$

---

집단간 분석
1. 가설 설립
$$H_0 : \mu_1 = \mu_2 = \mu_3 = \mu_4$$

$$H_1 : not \space \space H_0$$

2. P-value = 0.001 $$\alpha = 0.01$$

3. 검정 결과_통계적 결론
$$ H_0 \space \space reject $$

4. 결론 : 열처리 온도에 따른 강도의 차이가 존재한다.

---

신뢰구간 구하기

In [5]:
import math

In [14]:
Yb = 15
t2a = 2.179
MSE = 52.167
m = 4

In [15]:
llim = Yb - t2a*math.sqrt(MSE)/math.sqrt(m)
hlim = Yb + t2a*math.sqrt(MSE)/math.sqrt(m)

In [16]:
display(llim,hlim)

7.1308981635926205

22.86910183640738

In [17]:
def lh_lim(Yb,t2a,MSE,m):
    llim = Yb - t2a*math.sqrt(MSE)/math.sqrt(m)
    hlim = Yb + t2a*math.sqrt(MSE)/math.sqrt(m)
    return display(llim,hlim)

In [22]:
print('=============================')
# 125도 신뢰구간
lh_lim(15,2.179,52.167,4)
print('=============================')
# 150도 신뢰구간
lh_lim(24,2.179,52.167,4)
print('=============================')
# 175도 신뢰구간
lh_lim(26,2.179,52.167,4)
print('=============================')
# 200도 신뢰구간
lh_lim(43,2.179,52.167,4)
print('=============================')

7.1308981635926205

22.86910183640738

16.13089816359262

31.86910183640738

18.13089816359262

33.86910183640738

35.13089816359262

50.86910183640738

In [23]:
def lh_lim2(Yb1,Yb2,t2a,MSE,l):
    llim = (Yb1 - Yb2) - t2a*math.sqrt(2*MSE)/math.sqrt(l)
    hlim = (Yb1 - Yb2) + t2a*math.sqrt(2*MSE)/math.sqrt(l)
    return display(llim,hlim)

In [37]:
lh_lim2(15,24,2.179,52.167,4)

-20.128590540742344

2.128590540742346

In [29]:
def lh_lim3(Yb,t,s,n):
    llim = Yb - t * s / math.sqrt(n)
    hlim = Yb + t * s / math.sqrt(n)
    return display(llim,hlim)

In [32]:
print('=============================')
lh_lim3(15,3.182,2.160,4)
print('=============================')
lh_lim3(24,3.182,5.164,4)
print('=============================')
lh_lim3(26,3.182,12.193,4)
print('=============================')
lh_lim3(43,3.182,5.354,4)
print('=============================')

11.56344

18.43656

15.784076

32.215924

6.600937000000002

45.399063

34.481786

51.518214

쓸모 있는 신뢰구간인가 -> 아니다. 이런건 아무 의미 없다. 두 집단간의 차이에 관한 신뢰구간이 더 쓸모 있다.

---

p.46 예제 2-2

In [47]:
A = [8.0,8,7,9,8,7,9,8,6,8,7,6]
B = [6.0,5,7,7,6,7,6,8,7,6,8,7]
C = [8.0,8,8,5,6,7,6,7,7,6,7,8]
D = [9,4.0,6,5,4,4,5,6,7,5,8,4]

In [48]:
import pandas as pd

In [49]:
ex2_2 = pd.DataFrame({'A' : A, 'B' : B, 'C' : C, 'D' : D})
ex2_2

A    B    C    D
0   8.0  6.0  8.0  9.0
1   8.0  5.0  8.0  4.0
2   7.0  7.0  8.0  6.0
3   9.0  7.0  5.0  5.0
4   8.0  6.0  6.0  4.0
5   7.0  7.0  7.0  4.0
6   9.0  6.0  6.0  5.0
7   8.0  8.0  7.0  6.0
8   6.0  7.0  7.0  7.0
9   8.0  6.0  6.0  5.0
10  7.0  8.0  7.0  8.0
11  6.0  7.0  8.0  4.0

$$R : F >= 2.816$$

In [50]:
ex2_2.describe()

A          B          C          D
count  12.000000  12.000000  12.000000  12.000000
mean    7.583333   6.666667   6.916667   5.583333
std     0.996205   0.887625   0.996205   1.676486
min     6.000000   5.000000   5.000000   4.000000
25%     7.000000   6.000000   6.000000   4.000000
50%     8.000000   7.000000   7.000000   5.000000
75%     8.000000   7.000000   8.000000   6.250000
max     9.000000   8.000000   8.000000   9.000000